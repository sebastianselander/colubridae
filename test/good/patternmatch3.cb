// declare an algebraic data type
type Foo {
    Bar,
    Foo,
    Baz(int, string, int, int, bool),
}

def main() {
    let var = Baz(420, "hej", 69, 1337, false);
    // pattern match
    let x = match var {
        Baz(n,str, m, k, b) => {
            printInt(n);
            printString("\n");
            printString(str);
            printString("\n");
            printInt(m);
            printString("\n");
            printInt(k);
            printString("\n");
            printBool(b);
            printString("\n");
        }
    };
}
 
def printBool(b: bool) -> () {
    if b {
        printString("true")
    } else {
        printString("false")
    }
}
