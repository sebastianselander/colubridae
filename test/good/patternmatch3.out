420
hej
69
1337
false
